src/cache/lru.c

```diff
@@ -114,15 +114,14 @@ struct _hash {
   size_t size;
 };
 
-static inline size_t roundup2(size_t s) {
+static inline unsigned long long roundup2(unsigned long long s) {
   s--;
   s |= s >> 1;
   s |= s >> 2;
   s |= s >> 4;
   s |= s >> 8;
   s |= s >> 16;
-  if (sizeof(size_t) >= 8)
-    s |= s >> 32;
+  s |= s >> 32;
```

Comment:
Not in a truly portable manner... GCC offers
```
> gcc -E -dM -m64 - < /dev/null | grep __SIZEOF
#define __SIZEOF_INT__ 4
#define __SIZEOF_POINTER__ 8
#define __SIZEOF_LONG__ 8
#define __SIZEOF_LONG_DOUBLE__ 16
#define __SIZEOF_SIZE_T__ 8
#define __SIZEOF_WINT_T__ 4
#define __SIZEOF_PTRDIFF_T__ 8
#define __SIZEOF_FLOAT__ 4
#define __SIZEOF_SHORT__ 2
#define __SIZEOF_INT128__ 16
#define __SIZEOF_WCHAR_T__ 4
#define __SIZEOF_DOUBLE__ 8
#define __SIZEOF_LONG_LONG__ 8
```

But MSVC doesn't offer those or any equivalents that I know. We could enumerate 64-bit architectures, like `__x86_64__`, but that list would be fragile, would never be exhaustive and could vary by compiler.

I would strongly prefer the `unsigned long long` solution. C99 requires it to be at least 64 bits wide, and even the one heretic compiler we care about that doesn't officially support C99 does support `unsigned long long` and does make it 64-bits wide.